Panorama Public updates and fixes

vagisha opened the pull request:
#### Changes
- Include text and link to add a catalog entry in the Panorama Public notification message.
- Assign PXD accession and DOI at the end of CopyExperimentFinalTask so that we don't create these identifiers again in case the task has to be rerun due to a previous error.
- Fix NPE in confirmPublish.jsp if data license is null.
- Fix NPE if the dataVersion of an experiment is null. This can happen if there is an error in the data copy pipeline job.
- Log names of users that were assigned Reader role in the Panorama Public copy. Always include the form submitter and users assigned as data submitter and lab head. 
- Added more file audit events. Set directory and file on audit event.  Fixed container path in log messages.
- When a Panorama Public container is deleted look at all containers and sub-containers that may have a symlink target in the container being deleted.

vagisha committed:
- In the Panorama Public notification message, include text and link to add a catalog entry.
- Assign PXD accession and DOI at the end of CopyExperimentFinalTask so that we don't create these identifiers again in case the task has to be rerun due to a previous error.
- Fix NPE in confirmPublish.jsp if data license is null.
- Fix NPE if the dataVersion of an experiment is null. This can happen if there is an error in the data copy pipeline job.
- Log names of users that were assigned Reader role in the Panorama Public copy.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/view/publish/confirmPublish.jsp:
```diff
@@ -32,11 +33,8 @@
-            if (<%= bean.getLicense() != null %>) {
-                html += ' It will be available under the ' + <%=qh(bean.getLicense().getDisplayName())%> + ' license.';
-            }
+            html += <%= bean.getLicense() != null ? qh("It will be available under the " + bean.getLicense().getDisplayName() + ".") : HtmlString.EMPTY_STRING %>;
```
Maybe I'm misreading the code, but if the license is null, I think this will render as:

`html += ;`

which isn't valid JS.

vagisha replied:
Indeed.  It has been fixed. Thanks for catching.

vagisha committed:
Fix invalid JS.

vagisha committed:
- Set directory and file for audit event. Updated comment text.
- Added more file audit events.
- Fix container path in log messages.
- In fireSymlinkCopiedExperimentDelete() callhandleAllSymlinks() instead of handleContainerSymlinks() so that moving files back to sub-folders of the experiment folder are also handled.

vagisha committed:
Updated comments

vagisha committed:
Fixed failing test.

labkey-jeckels reviewed (approved):
The updates look good.

vagisha committed:
- Updated catalog entry notification text.
- Save ExperimentAnnotation to database after assigning PXD and DOI. Remove DOI from old copy after assigning it to the new copy.
